example/cpp-example/cronos.cc

```diff
@@ -493,6 +504,92 @@ void test_erc20_total_supply() {
   assert(total_supply == u256("100000000000000000000000000000000"));
 }
 
+// sample code for calling smart-contract
+void test_dynamic_api_send() {
+  std::ifstream t("../../common/src/contract/erc721-abi.json");
+  std::stringstream buffer;
+  buffer << t.rdbuf();
+  std::string json = buffer.str();
+
+  String mymnemonics = getEnv("MYMNEMONICS");
+  String mycronosrpc = getEnv("MYCRONOSRPC");
+  String mycontract = getEnv("MYCONTRACT721");
+  int mychainid = stoi(getEnv("MYCRONOSCHAINID").c_str());
+  Box<Wallet> mywallet = createWallet(mymnemonics);
+
+  String senderAddress = mywallet->get_eth_address(0);
+  String receiverAddress = mywallet->get_eth_address(2);
+  auto thisNonce = get_eth_nonce(senderAddress.c_str(), mycronosrpc);
+  cout << "rpc=" << mycronosrpc << endl;
+  std::string tokenid;
+  std::cout << "Enter tokenid: ";
+  std::cin >> tokenid;
+
+  Box<EthContract> w = new_eth_contract(json);
+
+  char tmp[300];
+  memset(tmp, 0, sizeof(tmp));
+  sprintf(tmp,
+          "[{\"Address\":{\"data\":\"%s\"}},{\"Address\":{\"data\":\"%s\"}},{"
+          "\"Uint\":{\"data\":\"%d\"}}]",
+          senderAddress.c_str(), receiverAddress.c_str(), stoi(tokenid));
+  std::cout << tmp << std::endl;
+  std::string paramsjson = tmp;
+  Vec<uint8_t> data; // encoded
+  data = w->encode(mycronosrpc, mycontract, "safeTransferFrom", paramsjson);
+  cout << "data length=" << data.size() << endl;
+  char hdpath[100];
+  int cointype = 60;
+  int chainid = mychainid; // defined in cronos-devnet.yaml
+  snprintf(hdpath, sizeof(hdpath), "m/44'/%d'/0'/0/0", cointype);
+  Box<PrivateKey> privatekey = mywallet->get_key(hdpath);
+  EthTxInfoRaw eth_tx_info = new_eth_tx_info();
+  eth_tx_info.to_address = mycontract;
+  eth_tx_info.nonce = thisNonce;
+  eth_tx_info.amount = "0";
+  eth_tx_info.amount_unit = EthAmount::EthDecimal;
+  eth_tx_info.data = data;
+  eth_tx_info.gas_limit = "219400";
+  eth_tx_info.gas_price = "100000000";
+  eth_tx_info.gas_price_unit = EthAmount::WeiDecimal;
+
+  Vec<uint8_t> signedtx =
+      build_eth_signed_tx(eth_tx_info, chainid, true, *privatekey);
+  CronosTransactionReceiptRaw receipt =
+      broadcast_eth_signed_raw_tx(signedtx, mycronosrpc, 1000);
```

Comment:
it's still helpful to have it split like written here, so that one can send the unsigned encoded tx for walletconnect or other signers.

but I was wondering if `EthContract` could have a `send` method that will do all the steps (signing and broadcast), similar to https://github.com/crypto-com/defi-wallet-core-rs/blob/main/bindings/cpp/src/contract.rs#LL67 ?